pkg/metrics/metric_collector.go

```diff
@@ -0,0 +1,249 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	autoscalingapi "github.com/gocrane/api/autoscaling/v1alpha1"
```

Comment:
Please format the imports by https://github.com/gocrane/crane/blob/main/docs/code-standards.md